wasi/wasi.go

```diff
@@ -277,16 +323,28 @@ func (w *WASIEnvironment) fd_close(ctx *wasm.HostFunctionCallContext, fd uint32)
 	return ESUCCESS
 }
 
-func args_sizes_get(ctx *wasm.HostFunctionCallContext, argcPtr uint32, argvPtr uint32) (err Errno) {
-	// not implemented yet
-	binary.LittleEndian.PutUint32(ctx.Memory.Buffer[argcPtr:], 0)
-	binary.LittleEndian.PutUint32(ctx.Memory.Buffer[argvPtr:], 0)
-	return 0
+func (w *WASIEnvironment) args_sizes_get(ctx *wasm.HostFunctionCallContext, argsCountPtr uint32, argsBufSizePtr uint32) Errno {
+	if !ctx.Memory.PutUint32(argsCountPtr, uint32(len(w.args.strings))) {
+		return EINVAL
+	}
+	if !ctx.Memory.PutUint32(argsBufSizePtr, w.args.totalBufSize) {
+		return EINVAL
+	}
+
+	return ESUCCESS
 }
 
-func args_get(*wasm.HostFunctionCallContext, uint32, uint32) (err Errno) {
-	// not implemented yet
-	return
+func (w *WASIEnvironment) args_get(ctx *wasm.HostFunctionCallContext, argsPtr uint32, argsBufPtr uint32) (err Errno) {
+	if !ctx.Memory.ValidateAddrRange(argsPtr, uint64(len(w.args.strings))*4) || !ctx.Memory.ValidateAddrRange(argsBufPtr, uint64(w.args.totalBufSize)) {
+		return EINVAL
+	}
+	for _, arg := range w.args.strings {
+		binary.LittleEndian.PutUint32(ctx.Memory.Buffer[argsPtr:], argsBufPtr)
+		argsPtr += 4
```

Comment:
trailing comment any magic numbers and they lose their magic!
```suggestion
		argsPtr += 4 // size of uint32
```

Reply:
I've introduced a file to define the sizes of types like `uint32`. This is in `. /wasi/type_size.go` for now, but we may want to move it to a more general directory later, since we had tended to use magic values like `4` to indicate the size of a type everywhere, and we may want to refactor those as well. JFYI @mathetake

Reply:
I believe what @codefromthecrypt means is not to add a file and define consts for this, but rather just add the comment. It doesn't matter whether the existing code base has comments or not. Having consts and user everywhere doesn't improve anything because the callsites of these consts still needs a comment.

Reply:
yep that's what I meant. I can try to be more clear later, but I also use trailing comments a lot to remind what it is as ex not everything is fixed encoding either